### PR TITLE
Clockwork Cult Defenses Patch - Leader role, traps and wiring, hulk and mech soft-counters, +more

### PR DESCRIPTION
**Pull request by ghost**
Hey, the freeze *is* over!
@Armhulen If you can get me those sprites, I can work to get the brazen bolter into this.

:cl: Xhuis
add: Added the Eminence role to clockcult! Players can elect themselves or ghosts as the Eminence from the eminence spire structure on Reebe.
add: The Eminence is incorporeal and invisible, and directs the entire cult. Anything they say is heard over the Hierophant network, and they can issue commands by middle-clicking themselves or different turfs.
add: The Eminence also has a single-use mass recall that warps all servants to the Ark chamber.
add: Added traps, triggers, and brass filaments to link them. They can all be constructed from brass sheets, and do different things and trigger in different ways. Current traps include the brass skewer and steam vent, and triggers include the pressure sensor, lever, and repeater.
add: The Eminence can activate trap triggers by clicking on them!
add: Servants can deconstruct traps instantly with a wrench.
del: Mending Mantra has been removed.
tweak: Clockwork scriptures have been recolored and sorted based on their functions; yellow scriptures are for construction, red for offense, blue for defense, and purple for niche.
balance: Servants now spawn with a PDA and black shoes to make disguise more feasible.
balance: The Eminence can superheat up to 20 clockwork walls at a time. Superheated walls are immune to hulk and mech punches, but can still be broken conventionally.
balance: Clockwork walls are slightly faster to build before the Ark activates, taking an extra second less.
fix: Poly no longer continually undergoes binary fission when Ratvar is in range.
code: The global records alert for servants will no longer display info that doesn't affect them since the rework.
/:cl:

I've been hearing a lot of complaints about the lack of depth and ingenuity in base building, as well as an inability to coordinate. During the freeze, I took these into account, and added this. Though most of the changes are detailed in the changelog, here's some of the stuff I glossed over to prevent a massive changelog body:
* The brass skewer impales any carbon mobs standing on it when it goes off. This hoists them into the air dealing 20 damage and makes them slowly bleed out while impaled. Impaled people can hold and use items normally but can't destroy their own skewer.
  * It takes seven seconds for someone to pull you off of a skewer, or you can resist off with thirty.
* Steam vents do no damage on their own, but block vision when active. Passing through them also makes you slightly wet because of the steam.
* Pressure sensors are difficult to see, and activate when a non-servant steps on it. They also auto-link with any traps they're built on.
* Levers activate when pulled. Non-servants can also activate them, since pulling a lever is not difficult for most people with a functioning brain.
* Repeaters activate every second when toggled on.
* The Eminence can warp to and from the station freely as long as the Ark exists, and can move around and scout. They *can't* understand normal speech unless the herald's beacon is activated.
* When Ratvar is summoned, the Eminence becomes visible and leaves a large swath of proselytized tiles behind it, and anything it says is heard by people like normal speech.
* Changed the command intercept for clockcult, since the old one wasn't as relevant anymore.
* Allowed any mob to become a servant, not just living ones. This is to support the Eminence, but there are proper checks to prevent different mobs that shouldn't be servants from becoming them.
* Also includes the manacle sprites by @FuryMcFlurry for Hateful Manacles.
* Added a file in the `strings` folder called `clockwork_cult_changelog.txt` - the Ark primer that servants get on roundstart draws from this file to build its list of changes.
* Removed Mending Mantra. I've bee turning over in my head how a powerful AoE heal could be balanced, but in the end I decided that any costs would be too crippling if one person spammed it and other solutions would just confuse people who are used to the old version.
* Changed the judicial visor's sound to the distant explosion instead of the far once since it's fancier!

Thoughts, feedback, ideas, all that jazz are all welcome and encouraged. I'm still considering changing the cost of building walls, but having done some local testing, building a lot of stargazers is both fast and easy - and now, with traps, much easier to defend and conceal - and generating the power for a large defense seems easy on paper if the Eminence coordinates effectively!

I'd like this to be testmerged a bit so I can work off of feedback before the final merge.

### CODE REVIEW

**Comment by Xhuis** [comment]
I think @IconDiffBot is still having aneurysms.

**Comment by Arianya** [comment]
I'm uncertain because IconDiffBot keeps having seizures, but I presume super-heated walls look identical to normal walls and are only identifiable by trying to punch them?

**Comment by Xhuis** [comment]
Superheated walls have a very distinct red glow. It changes the color of the wall instead of using a different icon, but it's very easy to see if a wall is superheated at just a glance.

**Comment by Arianya** [comment]
I'd be inclined to lean away for that, since otherwise hulks (who tend to have x-ray by virtue of genetics gone wild) will be able to very rapidly spot where the non-superheated walls are and aim for that "weakpoint" to the exclusion of all else. I'd be inclined to either examine text or simply special text when a hulk/mech punches a superheated wall

**Comment by GuyonBroadway** [comment]
You coulda made the Eminence svetung just as a thought. 

It'd make sense for one of Ratvars generals to be leading the fight. I know that Uncle-proton has sprites for it. (or at least very good even if WiP sprites)

**Comment by Xhuis** [comment]
Sevtug is the most apathetic of the generals and would probably spend the whole shift torturing people on the station for amusement. If I *did* use any of the generals, I think it would likely be Nezbere.

**Comment by GuyonBroadway** [comment]
Actually on that note it would be interesting if you could "choose" the general who becomes the Eminence each with their own abilities. 

Might be a bit beyond this PR's scope but it would be nice to have a tactical choice.

**Comment by DoctoreMad** [comment]
Nice buffs where are the nerfs

**Comment by Xhuis** [comment]
I think the cult with I estimate a 20% or so winrate from a brief glance at statbus doesn't need any further nerfs.

**Comment by Arianya** [comment]
@DoctoreMad They drop from arcade machines when you win

**Comment by DoctoreMad** [comment]
**ALRIGHT I GUESS**

**Comment by Shadowflame909** [comment]
So the eminence is the leader? Hand of god confirmed!?!?!

**Comment by davethwave** [comment]
so borgs and the like can now be converted to the clock work side?

**Comment by Xhuis** [comment]
Silicons have always been eligible to be converted, even since day one of the gamemode itself.

**Comment by DoctoreMad** [comment]
@davethwave Play the gaaaameeeeeee

**Comment by davethwave** [comment]
I was under a misconception that that feature was was removed. Over the time when reebe base started I've never seen a Borg get converted with a rune of submission since.

File: code/game/gamemodes/clock_cult/clock_helpers/scripture_checks.dm [hunk 1]
```diff
@@ -43,7 +43,9 @@
 
 //changes construction value
 /proc/change_construction_value(amount)
-	GLOB.clockwork_construction_value += amount
+	if(!SSticker.current_state != GAME_STATE_PLAYING) //This is primarily so that structures added pre-roundstart don't contribute to construction value
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_helpers/scripture_checks.dm]: construction value doesn't do anything since the rework, though.

File: code/game/gamemodes/clock_cult/clock_structures/eminence_spire.dm [hunk 2]
```diff
@@ -0,0 +1,140 @@
+//Used to nominate oneself or ghosts for the role of Eminence.
+/obj/structure/destructible/clockwork/eminence_spire
+	name = "eminence spire"
+	desc = "A hulking machine made of powerful alloy, with three small obelisks and a huge plate in the center.."
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_structures/eminence_spire.dm]: double period

File: code/game/gamemodes/clock_cult/clock_structures/traps/brass_skewer.dm [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+		return
+	var/mob/living/carbon/squirrel = locate() in get_turf(src)
+	if(squirrel)
+		squirrel.visible_message("<span class='boldwarning'>A massive brass spike erupts from the ground, impaling [squirrel]!</span>", \
+		"<span class='userdanger'>A massive brass spike rams through your chest, hoisting you into the air!</span>")
+		squirrel.emote("scream")
+		playsound(squirrel, 'sound/effects/splat.ogg', 50, TRUE)
+		playsound(squirrel, 'sound/misc/desceration-03.ogg', 50, TRUE)
+		squirrel.apply_damage(20, BRUTE, "chest")
+		squirrel.pixel_y = 8
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_structures/traps/brass_skewer.dm]: pretty sure that this isn't the proper way to handle buckling pixel offsets; see how rollerbeds do it

File: code/game/gamemodes/clock_cult/clock_structures/traps/brass_skewer.dm [hunk 4]
```diff
@@ -0,0 +1,88 @@
+//Non-servants standing over this will get spikes through the feet, immobilizing them until they're freed.
+/obj/structure/destructible/clockwork/trap/brass_skewer
+	name = "brass skewer"
+	desc = "A massive brass spike, cleverly concealed in the floor. You think you could prevent setting it off if you moved slowly."
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_structures/traps/brass_skewer.dm]: The second part is more a reference to how pressure sensors work, though they're probably the best way to trigger these.

File: code/game/gamemodes/clock_cult/clock_helpers/scripture_checks.dm (hunk 1 above)
Comment (Xhuis) [reply code/game/gamemodes/clock_cult/clock_helpers/scripture_checks.dm]: I had some code for wall costs scaling with CV in, but it didn't work out; I must have missed this when I was gutting what I had before

File: code/game/gamemodes/clock_cult/clock_structures/traps/brass_skewer.dm [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+		playsound(squirrel, 'sound/misc/desceration-03.ogg', 50, TRUE)
+		squirrel.apply_damage(20, BRUTE, "chest")
+		squirrel.pixel_y = 8
+		buckle_mob(squirrel, TRUE)
+	else
+		visible_message("<span class='danger'>A massive brass spike erupts from the ground!</span>")
+	playsound(src, 'sound/machines/clockcult/brass_skewer.ogg', 75, FALSE)
+	icon_state = "[initial(icon_state)]_extended"
+	density = TRUE //Skewers are one-use only
+	desc = "A massive brass spike protruding from the ground like a snapped bone. It makes you sick to look at."
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_structures/traps/brass_skewer.dm]: this description is weird, since it's like, not broken at this point? it's literally perfectly fine.

File: code/game/gamemodes/clock_cult/clock_structures/traps/brass_skewer.dm [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+		return L.bullet_act(P)
+	return ..()
+
+/obj/structure/destructible/clockwork/trap/brass_skewer/activate()
+	if(density)
+		return
+	var/mob/living/carbon/squirrel = locate() in get_turf(src)
+	if(squirrel)
+		squirrel.visible_message("<span class='boldwarning'>A massive brass spike erupts from the ground, impaling [squirrel]!</span>", \
+		"<span class='userdanger'>A massive brass spike rams through your chest, hoisting you into the air!</span>")
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_structures/traps/brass_skewer.dm]: you need to do something like how I did geis bindings, so that the spike actually goes through their chest.

**Comment by Ratvar** [comment]
Let Eminence use TK on War Clockcult?

**Comment by Xhuis** [comment]
That seems like it might be way too po-
Wait, you're Ratvar. I guess I kind of have to do that.

**Comment by AdamElTablawy** [comment]
Do pressure plates and such show above, say, tables? Because the way I see it I see table checkerboard hell again, but with pressure plates below them and a skewer under it. Climbing/smashing through tables with sigils on top is already hell, I don't want to imagine table sigil hell but there are fucking skewers on it and the warden is zapping you the entire time

IE as long as I can build multiple objects on the same tile, that's going to be a problem. Also, I see steam vents being hell, honestly. If you can't see where the warden is that's zapping you and there are skewers fucking everywhere, ESPECIALLY if you can put a steam vent on top of a skewer and such, then clockcult is going to be even more impossible for normal crew to get through.

In fact, all this seems to do is make it so that, once again, normal crew cannot, at all, get through reebe because it's hell, and the only things that CAN are mecha and hulks, of which the only thing that can actually annoy them is the new Eminence. 

So in summary makes it even more impossible on ordinary crew because now the stun sigils are stun sigil v2: meatspike hell edition and you're going to have the same nonsense tables-on-top-of-windows-with-a-girder-in-the-table, without doing anything to address the actual problem of virtually needing a mecha or an xray laser to get through reebe or you won't make it past the first three lines of a decent defense.

**Comment by Xhuis** [comment]
Ocular wardens cannot see through steam vents. Setting up skewers on every tile is massively time-consuming and will likely buy the crew the time to gather the resources needed to break through anyway; smashing pressure sensors is nearly instant as they only have 5 integrity and alt-clicking a table is the only thing you need to do in order to see if there's a sensor under it or not. Once the sensor has been destroyed, the skewer is rendered useless unless they set up a backup lever or some sort.

So, yes, blindly rushing in will indeed result in you being impaled on several spikes and likely bleeding out on the top of one by the time you're dead. Blindly rushing in has always been a death sentence against a competent cult, because it requires some semblance of thought instead of grabbing the nearest tool and charging in to get your valids on.

**Comment by AdamElTablawy** [comment]
How long is skewer build time? Because defenses can actually get built very quickly if the cult cooperates.

Aside from that, it's less a matter of 'blindly rushing in and not checking for traps' and more 'I was moving backwards to avoid this ocular warden or random marauder charging at me or some idiot walked with harm intent on, and now I am impaled and almost guaranteed to die outright'. Essentially, it would be a problem more at the back defenses where most of the cultists usually are. The crew breaks through like, what, the tenth line of tables-with-girders-on-them-and-sigils-everywhere and inevitably gets into the 'cult-fighting-the-crew' clusterfuck, but this time there are traps everywhere which means movement in combat has the sudden risk of being fucking impaled out of nowhere, which is way worse than just stepping on a stun sigil because with a stun sigil they can drag you out of the way.

And again, people really just can't alt click every tile to see if there's something there, especially if the skewers are just randomly distributed. You run across a tile where someone's dropped something and whoops, there's a fucking skewer there out of the blue. None of the other tiles had that, you were just unlucky, and unlike with sigils, you can't actually see the sprite all too well.

Do steam vents or anything cost power to keep running? Or can they just constantly disable vision. Because while ocular wardens may not work I forsee many nightmares with steam vents and mania motors.

How about actually being able to use the brass filaments to turn on mania motors? That might make them marginally more usable rather than either having them constantly running and being a huge power drain and or virtually requiring them to be at the back so servants can turn them on.

**Comment by Nabski89** [comment]
Wait, what's wrong with making slimes and xenos into cultists?

**Comment by Xhuis** [comment]
Skewers take either two or three seconds to build. Pressure sensors take an additional three (or whatever you want to use to trigger them) and wiring them up with filaments depends on how good you are at clicking tiny sprites.

If you walk into a skewer because you saw an item and went for it, that's you triggering the trap. Mania motors also don't work through steam vents because they only work on people who can see them. As I said before, they block *all* vision. This doesn't discriminate at all. It uses the same variable that walls do. If a wall would block something based on sight, then so will an active steam vent. (You have to wire those up too.)

Currently, yes, you can stack traps on one tile. I actually attempted to find a workaround for this but because of byond being awful I couldn't find a way to do so since `parent_type` is both simultaneously unable to be run through `initial()` and unable to be run without `initial()` for an uninitialized typepath like every single other variable it has can be.

There remain plenty of ways to circumvent these things as you are viewing them in vacuum. Silicons, simple animals, mechs, and the like all don't set off skewers. Oh, and neither does walking. If you stop running, skewers can't be set off by stepping on the pressure sensors. Just a point of note!

@Nabski89 It could work and be rather silly, but I'd have to think about the repercussions of it first.

**Comment by Xhuis** [comment]
When I push the next commit, a skewer with a pressure sensor will look like this:

**Comment by Armhulen** [comment]
This is a really good step in the right direction, conversion is next >:)

**Comment by Armhulen** [comment]
Also skewers should be VISIBLE but take no damage until extended.

**Comment by Xhuis** [comment]
Skewers can be smashed while still retracted; not having that would be somewhat inconsistent.

**Comment by AdamElTablawy** [comment]
Truthfully if a cultist manages to convert a xenomorph that is an achievement in and of itself and they fully deserve it.

I didn't mean 'saw an item and went for it', I mean 'some asshole dropped their box or an oxygen tank on the floor and you didn't see the pressure plate because of it because some random asshole on your side dumped shit from their inventory to make room for a bottle of fucking eggnog or something'.

Alright, didn't know mania motors didn't work on people without direct vision to it. 

Pressure sensors are so invisible that I can barely see them. What's the skewer sprite look like pre-activation?

3 seconds and 2 brass is fairly cheap, all things considered.

**Comment by Xhuis** [comment]
You say that the cult is unbeatable, but the publically-visible stats on statbus show a huge loss/win ratio for clockwork cult. If the average base becomes unbreakable enough that it needs to addressed, I'd certainly do some nerfs, but as it stands they seem to need some fairly large buffs as it stands. There remains ten minutes (not including the grace period for the crew to arm up!) between the Ark starting and its activation, and that is more than enough time to break down every table, window, and girder in Reebe - doubly so if the crew properly armed up in the grace period.

File: code/game/gamemodes/clock_cult/clock_helpers/scripture_checks.dm (hunk 1 above)
Comment (Xhuis) [reply code/game/gamemodes/clock_cult/clock_helpers/scripture_checks.dm]: I do think I'll keep this in for future work, however; maybe even in this PR. It also makes more sense, and because CV is announced at round end, it gives players a picture of how large the base was.

**Comment by DoctoreMad** [comment]
You know this publically visible stats stuff is annoying when you consider the amount of Clockwork cults that lose solely because one side or another is retarded.

**Comment by GunHog** [comment]
> You know this publically visible stats stuff is annoying when you consider the amount of Clockwork cults that lose solely because one side or another is retarded.

But, when incompetence is so common that it is essentially the essence of our playerbase, can you really dismiss it?

File: code/game/gamemodes/clock_cult/clock_structures/traps/brass_skewer.dm [hunk 7]
```diff
@@ -49,20 +51,31 @@
...
+	desc = "A vicious brass spike protruding from the ground like a stala[pick("gm", "ct")]ite. It makes you sick to look at." //is stalagmite the ground one? or the ceiling one? who can ever remember?
 
 /obj/structure/destructible/clockwork/trap/brass_skewer/user_buckle_mob()
 	return
 
+/obj/structure/destructible/clockwork/trap/brass_skewer/post_buckle_mob(mob/living/L)
+	if(L in buckled_mobs)
+		L.pixel_y = 3
+		impale_overlay = mutable_appearance('icons/obj/clockwork_objects.dmi', "brass_skewer_pokeybit", ABOVE_MOB_LAYER)
+		L.add_overlay(impale_overlay)
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_structures/traps/brass_skewer.dm]: it'd be much better for this to be an overlay on the skewer itself rather than the mob, and I don't see any reason why it can't be.

**Comment by DoctoreMad** [comment]
@GunHog 
If that's true, we're in a truly messy stage right now. To fix the problem of unskilled cultists, we're adding tools those _PEOPLE_ can't even use-- And I know this when I see that the lines of defense are lax even though we have gathered energy for the sake of arming up Reebe. So, if a cogscarab says, "Build me a belligerent eye", this is a very serious order and easy to follow, but none of them stop doing it.
I do not know what kind of problem they have, but it is what it is.

**Comment by Shadowflame909** [comment]
Honestly, the main problem with clock cult losing is that they focus on on station missions instead of base building. Sure, you can spend all your time and efforts taking down tcomms, and building stargazers. But who's making stuns on the base? The drones can only make walls..and that's not enough most of the time. Unless the Clock Cult converted 99 percent of the crew.

**Comment by AdamElTablawy** [comment]
The clock cult, if the crew does not have mechs, and they have set up a decent defense, is effectively unbeatable for normal crew because there are just so many obstacles everywhere that trying to move through them before the timer runs out is an exercise in futility.

Most clock cults lose because their members just go braindead, warp to station and die, or neglect to set up any defenses or don't stop robotics from making a mech. This is why clock cult has a 20% winrate: A combination of incompetence on their side, and the fact that mechs are both so useful and so vital for a crew victory that they get built every time. A crew without a mech or an xray laser is almost virtually guaranteed to lose simply because the sheer number of defenses from a half competent cult means that even if the timer doesn't outright run out, the usual two-minute-clusterfuck once you finally breach the last line of defense will sap the rest of your time away and you lose. This will exaggerate this problem.

The cult definitely needs a buff. But it doesn't need a buff against ordinary humans: it has enough defenses against those. In fact, it's probably way too good against ordinary humans right now. It definitely needs more against mecha, and mecha need to not be so fucking vital for ordinary humans that without one the crew is virtually guaranteed to eat shit and die.

**Comment by JJRcop** [comment]
@AdamElTablawy So, the solution is to allow mechs to be converted! (sparns a soul vessel in them)

**Comment by Arianya** [comment]
I think people need to realize that clockcult is never going to be something you can solo as a human. Its a team vs team effort and if you rush in and get skewered you are (largely) reliant on your team to rescue you and hopefully patch you up.

The times I've seen good, engaging clockcult rounds its been when the crew don't have mechs/hulks (or not many hulks, as it might be), when the crew are skirmishing from their protective barrier and the clock cults are repairing as they go, when one person getting downed creates a mad rush between both sides to either heal their team mate or to secure the kill.

Defenses as they currently exist are a good devaluer of numbers. A 5-10 man cult *can* hold off a 40 man crew with some decent teamplay, even without an endless maze of tablewalls.

Equally, a single "strike team" of crew can breach a weakpoint in the defenses and push through to the victory. The Ark itself has almost no HP so once its exposed, its in trouble.

The current imbalancers to this are hulks and mechs, and the meta stabilized to such a point that as soon as the slightest whiff of clockcult comes around, you see things like armed security/captains lounging around genetics/robotics to ensure that no sabotage can take place, and going loud as cultists is generally a terrible idea (for not bad reasons, they're meant to be insurgents, not a army that takes over the station pre-Ark activation).

So yes, these changes are good, especially because skewers work on hulks

**Comment by Rinoka** [comment]
On unmerge, the plant bug seems to be unrelated to this PR.

However, this PR seems to cause guns to permafire north. So that could use some fixing.

**Comment by Xhuis** [comment]
I tested that locally. This PR does not cause that issue.

File: code/__DEFINES/clockcult.dm [hunk 8]
```diff
@@ -1,9 +1,9 @@
...
-#define BELLIGERENT_EYE "belligerent_eye"
-#define VANGUARD_COGWHEEL "vanguard_cogwheel"
-#define GEIS_CAPACITOR "geis_capacitor"
+//component id defines; sometimes these may not make sense in regards to their use in scripture but important ones are bright
+#define BELLIGERENT_EYE "belligerent_eye" //Use this for offensive and damaging scripture!
+#define VANGUARD_COGWHEEL "vanguard_cogwheel" //Use this for defensive and healing scripture!
+#define GEIS_CAPACITOR "geis_capacitor" //Use this for niche scripture!
 #define REPLICANT_ALLOY "replicant_alloy"
-#define HIEROPHANT_ANSIBLE "hierophant_ansible"
+#define HIEROPHANT_ANSIBLE "hierophant_ansible" //Use this for construction-related scripture!
```
Comment (ChangelingRain) [inline code/__DEFINES/clockcult.dm]: this appears to be one lower than it should be.

File: code/game/gamemodes/clock_cult/clock_items/clockwork_slab.dm [hunk 9]
```diff
@@ -153,6 +160,11 @@
 		return 0
 	access_display(user)
 
+/obj/item/clockwork/slab/AltClick(mob/living/user)
+	if(is_servant_of_ratvar(user) && linking)
+		linking = null
+		to_chat(user, ",span class='notice'>Object link canceled.</span>")
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_items/clockwork_slab.dm]: \<

File: code/game/gamemodes/clock_cult/clock_items/clockwork_slab.dm [hunk 10]
```diff
@@ -410,7 +422,10 @@
 				data["tier_info"] = "<font color=#B18B25><i>Unlock these optional scriptures by converting another servant or if [DisplayPower(APPLICATION_UNLOCK_THRESHOLD)] of power is reached..</i></font>"
 
 	data["selected"] = selected_scripture
-
+	data["scripturecolors"] = "<font color=#DAAA18>Scriptures in <b>yellow</b> are related to construction and building.</font><br>\
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_items/clockwork_slab.dm]: it's pretty obvious, even now, that scriptures that used ansibles are long-ranged and power-related.

**Comment by Arianya** [comment]
merge this I wanna skewer peopleeee

**Comment by Xhuis** [comment]
I want to get at least one more playtest out so the removal of Mending Mantra can be tested live.

**Comment by ShizCalev** [comment]
@Xhuis #33114

**Comment by Xhuis** [comment]
Due to the way `hierophant_message` is, it sends that message to both servants and ghosts. I could separate them, if you want, but it's a tad silly.

**Comment by ChangelingRain** [comment]
`/proc/hierophant_message(message, servantsonly, atom/target)`
servantsonly will prevent the message from being seen by ghosts.

**Comment by Xhuis** [comment]
That's handy. 🤔

**Comment by imsxz** [comment]
Pardon if someone already suggested this, there's too many comments for me to read them all.

If you're not entirely against the idea of mending mantra existing, I have some nerf suggestions.

-draining vitality from the healing sigil things pool of vitality for every thing it heals
-making it damage the one channeling it (without healing themselves of course)
-change the ability to a once-every-while strong heal which takes a bit to channel, makes the one channeling glow/otherwise become obvious while doing it
-a combination of the above?

Personally, I believe the one strong heal on a long cooldown would be the best. Couple it with other effects such as making the user very obvious that he's doing it as to make the crew target him, a long channel time (5-10s). Maybe root the one casting it in place while casting it, making it a very high risk high reward spell.

This would add lots of counterplay as it would let the crew be able to quickly stop the healer if he's not well protected, and generally being a big risk.

**Comment by optimumtact** [comment]
recompile tgui

**Comment by Xhuis** [comment]
I'll recompile YOU

**Comment by Xhuis** [comment]
@imsxz The idea occurred to me. I tossed over or demoed a few different methods of mending mantra locally, like:
* A focused, single-target heal that healed for 10/second on a crewmember
* Beams between the healer and their allies so people can track them to the source
* Having a power cost on each chant of the invocation
* Scaling the power costs with various factors
* Needing to stand on a sigil of transmission to invoke it

In the end, they all had their own problems and upsides, but I feel that large, indefinite healing isn't balanced in this way. It was actually better-implemented in the mending motor structure before it was turned *into* mending mantra, and fit more with the reinforcement theme. 

Because it's available to everyone, can be canceled at any time, and doesn't cost power for continuous use, it's massively powerful, and even adding a huge + symbol over the person invoking it hasn't been enough to make people target them. Power is currently designed to either be in surplus or shortage, so if I make mending mantra take a ton of power it'd let people drain the entire supply without realizing and if it didn't drain much then it would have to be weak to compensate.

I'll have to give it some thought overall. In the meantime, I think I'm happy with this PR now!

**Comment by davethwave** [comment]
Perhaps as an idea use a health draining system for it instead with a semi weakish health drain and power drain(power from Clockwork power system) on enemies that heals allies nearby. If there are no enemies that are nearby have it use the player health pool instead with a larger power draw. But if there is no one nearby be it ally or enemy have it hurt the player by a certain amount of misc damage so someone doesn't use it to drain the power on purpose or have the health drain accelerate by a fair amount making it almost impossible to recover faster that you can heal via meds.

**Comment by Xhuis** [comment]
What I could replace it with is limited by the fact that anything that's remotely complex usually ends up never used because people don't put in the effort to understand it. Even if it *was* reasonably so, it would fall right back into the "arcane and obscure mechanics on everything" trap that old clockcult did.

**Comment by ChangelingRain** [comment]
It could use power for each thing it heals, like the motor used to.

**Comment by Xhuis** [comment]
I'm a bit concerned that that would lead to one of the aforementioned issues, of power being drained quickly because servants don't pay attention to the power drain. I think that if I would have it stay, it'd be in the form of the mending motor, with a decent power drain.

**Comment by AdamElTablawy** [comment]
Use vitality to heal people, then.

**Comment by Xhuis** [comment]
In case I wasn't obvious about it, this is ready. For the time being, I'm happy with this, so any more changes would come after observing based on what the changes end up doing.

File: code/game/gamemodes/clock_cult/clock_mobs/_eminence.dm [hunk 11]
```diff
@@ -0,0 +1,267 @@
...
+	move_on_shuttle = TRUE
+	see_in_dark = 8
+	invisibility = INVISIBILITY_OBSERVER
+	layer = FLY_LAYER
+	faction = list("ratvar")
+	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_INVISIBLE
+	var/static/superheated_walls = 0
+
+/mob/camera/eminence/Initialize()
+	if(SSticker.mode.eminence)
```
Comment (optimumtact) [inline code/game/gamemodes/clock_cult/clock_mobs/_eminence.dm]: I think you want to invert this check, otherwise you're gonna queue it for qdeletion when it inits (and then skip destruction due to it's Destroy check)

File: code/game/gamemodes/clock_cult/clock_mobs/_eminence.dm [hunk 12]
```diff
@@ -0,0 +1,267 @@
...
+		return INITIALIZE_HINT_QDEL
+	. = ..()
+
+/mob/camera/eminence/Destroy(force)
+	if(!force && SSticker.mode.eminence)
```
Comment (optimumtact) [inline code/game/gamemodes/clock_cult/clock_mobs/_eminence.dm]: Why is this here?

File: code/game/gamemodes/clock_cult/clock_scriptures/scripture_drivers.dm [hunk 13]
```diff
@@ -153,52 +244,9 @@
...
-	primary_component = GEIS_CAPACITOR
-	sort_priority = 6
-	quickbind = TRUE
-	quickbind_desc = "Creates a Sigil of Submission, which will convert non-Servants that remain on it."
-
-
 //Replicant: Creates a new clockwork slab.
 /datum/clockwork_scripture/create_object/replicant
-	descname = "New Clockwork Slab"
+	descname = "New Clockwork Slab - Important!"
```
Comment (optimumtact) [inline code/game/gamemodes/clock_cult/clock_scriptures/scripture_drivers.dm]: Important!

is that really needed?

File: code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm [hunk 14]
```diff
@@ -34,7 +37,21 @@
...
+/obj/structure/destructible/clockwork/massive/celestial_gateway/take_damage(damage_amount, damage_type = BRUTE, damage_flag = 0, sound_effect = 1, attack_dir)
+	. = ..()
+	if(.)
+		flick("clockwork_gateway_damaged", glow)
+		playsound(src, 'sound/machines/clockcult/ark_damage.ogg', 75, FALSE)
+		if(last_scream < world.time)
+			audible_message("<span class='boldwarning'>An unearthly screaming sound resonates throughout Reebe!</span>")
+			for(var/V in GLOB.player_list)
+				var/mob/M = V
+				if(M.z == z || is_servant_of_ratvar(M) || isobserver(M))
```
Comment (optimumtact) [inline code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm]: why do ghosts need to be subjected to this if they're off the z

File: code/game/gamemodes/clock_cult/clock_structures/eminence_spire.dm [hunk 15]
```diff
@@ -0,0 +1,140 @@
...
+			hierophant_message("<span class='brass'><b>No ghosts accepted the offer!</b> The eminence spire has been reset.</span>")
+			eminence_nominee = null
+			return
+		visible_message("<span class='warning'>A blast of white-hot light spirals from [src] in waves!</span>")
+		playsound(src, 'sound/machines/clockcult/ark_damage.ogg', 50, FALSE)
+		var/mob/camera/eminence/eminence = new(get_turf(src))
+		eminence_nominee = pick(candidates)
+		eminence.key = eminence_nominee.key
+		hierophant_message("<span class='bold large_brass'>A ghost has ascended into the Eminence!</span>")
+	for(var/V in GLOB.player_list)
```
Comment (optimumtact) [inline code/game/gamemodes/clock_cult/clock_structures/eminence_spire.dm]: you have like two or three places you do this, make it a macro or a proc

File: code/game/gamemodes/clock_cult/clock_scriptures/scripture_drivers.dm (hunk 13 above)
Comment (Xhuis) [reply code/game/gamemodes/clock_cult/clock_scriptures/scripture_drivers.dm]: It is; from my own experience people are braindead and constantly ask how to make new slabs, so with luck this'll make people do it more.

File: code/game/gamemodes/clock_cult/clock_mobs/_eminence.dm (hunk 12 above)
Comment (Xhuis) [reply code/game/gamemodes/clock_cult/clock_mobs/_eminence.dm]: I forgot a `== src` at the end there, whoops!

File: code/game/gamemodes/clock_cult/clock_mobs/_eminence.dm (hunk 11 above)
Comment (Xhuis) [reply code/game/gamemodes/clock_cult/clock_mobs/_eminence.dm]: `SSticker.mode.eminence` is set in Login, so this is intended to delete any new eminence mobs that somehow get made after one is already assigned.

File: code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm (hunk 14 above)
Comment (Xhuis) [reply code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm]: Ghosts hear everything on the hierophant network, and any other servant-specific sounds and messages, so I included this for consistency.

**Comment by Xhuis** [comment]
That should be all.

File: code/game/gamemodes/clock_cult/clock_mobs/_eminence.dm (hunk 11 above)
Comment (optimumtact) [reply code/game/gamemodes/clock_cult/clock_mobs/_eminence.dm]: ah okay that makes more sense now that the destroy case is fixed :+1:

**Comment by optimumtact** [comment]
Could you make your pr's more atomic, it's a wonder nobody reviews them

File: code/game/gamemodes/clock_cult/clock_cult.dm [hunk 16]
```diff
@@ -303,14 +315,19 @@ Credit where due:
 	<hr>\
 	<h2>Things that have changed:</h2>\
 	<ul>\
-	<li><b><i>Scripture no longer requires components, and instead uses power.</i></b></li>\
-	<li>Added a <b>5-minute grace period</b> for the crew to prepare for the assault when the Ark activates.</li>\
-	<li>Script and Application scriptures can now be unlocked with enough power.</li>\
-	<li><b>Added the Hateful Manacles scripture</b>, which handcuffs targets!</li>\
+	CLOCKCULTCHANGELOG\
```
Comment (optimumtact) [inline code/game/gamemodes/clock_cult/clock_cult.dm]: what is this

Comment (optimumtact) [reply code/game/gamemodes/clock_cult/clock_cult.dm]: nevermind i sread the rest

Comment (Xhuis) [reply code/game/gamemodes/clock_cult/clock_cult.dm]: OwO what's this?
